### PR TITLE
rgw: partial mesh

### PR DESCRIPTION
**Pull request by yehudasa**
Allows configuring partial mesh architecture within a single zone-group. By default it's still a full mesh, that is all zones sync from all other zones. With this change we can specify for each zone whether it's pulling data from all other zones, or which of the zones it'd be pulling the data from.

### CODE REVIEW

File: src/rgw/rgw_admin.cc [hunk 1]
```diff
@@ -1993,6 +2006,11 @@ int main(int argc, char **argv)
   std::string zonegroup_name, zonegroup_id, zonegroup_new_name;
   std::string api_name;
   list<string> endpoints;
+  int tmp_int;
+  int sync_from_all_specified = false;
```
Comment (cbodley) [inline src/rgw/rgw_admin.cc]: int -> bool

**Comment by idealguo** [comment]
Hi, this feature is important for multi clusters environment(in same city), How about the progress? Do we have chance to see it in next version?

**Comment by yehudasa** [comment]
@idealguo it's not going to be in kraken, planned for the version that follows it.

**Comment by yehudasa** [comment]
hmm, apparently all the changes in here already found themselves into master as part of the sync modules feature. Closing.

**Pull request closed without merging**